src/compat/compatcmdliner.ml

```diff
@@ -0,0 +1,28 @@
+(* Compatibility module for Cmdliner *)
+
+(* Cmdliner 1.1.0 has deprecated the 'traditional' Term and Arg modules,
+   but is only available for OCaml 4.08 and above. This compatibility
+   module will work on 1.0.4 and 1.1.0 for all supported versions of
+   the OCaml compiler without causing deprecation alerts *)
+
+[@@@ocaml.warning "-3"]
+module Term = struct
+  include Cmdliner.Term
+
+  type info2 = Cmdliner.Term.info
```

Comment:
Do you really have to rename the type ?

Reply:
Hrm, maybe not actually. Originally I just 'included' the whole of Cmdliner.Term/Arg, but that didn't remove the warnings, so then I just redeclared them, which did. Unfortunately 4.07 complained about having 2 types the same name, so I renamed. Instead I'll try getting rid of the include and just export a more minimal module.

Reply:
Also the `module Term = Cmdliner.Term [@ocaml.warning "-3"]` trick you linked to didn't work ?

Reply:
I see. That being said it seems the use of cmdliner is quite local. I wonder if it wouldn't be easier to disable the warnings after the `open Cmdliner` and call it a day. 

```
> git grep Cmdliner
doc/deps.mld:{2 Cmdliner}
doc/deps.mld:{!modules:Cmdliner}
src/odoc/bin/main.ml:open Cmdliner
src/odoc/bin/main.ml:  val extra_args : args Cmdliner.Term.t
test/odoc_print/odoc_print.ml:open Cmdliner
```